functorch/test/test_control_flow.py

```diff
@@ -0,0 +1,154 @@
+import torch
+
+from torch.testing._internal.common_utils import TestCase
+from functorch.experimental.cond import cond
+from torch.fx.experimental.proxy_tensor import make_fx
+
+
```

Comment:
We should add some test cases for the error cases (e.g. the branches return things that are different) for both tracing and regular execution

Reply:
Erm, I thought about this some more. For regular execution, cond should probably not check that the branches return different things? Because that would be unnecessary?

Reply:
It does not do so for regular execution - we only do so for tracing. Is that what you mean?

Reply:
Yes, we should add tests for the error cases for tracing. I was originally thinking that regular execution should also validate the error cases, but then changed my mind

Reply:
Done